### PR TITLE
feat: share ios logs in DebugView

### PR DESCRIPTION
**Pull request by cythb**

### CODE REVIEW

File: Maskbook/Scene/Debug/DebugView.swift [hunk 1]
```diff
@@ -13,11 +13,36 @@ struct DebugView: View {
...
+    private var coordinator: Coordinator
+    
     var body: some View {
-        VStack(alignment: .leading) {
+        VStack(alignment: .leading, spacing: 20) {
             NavigationLink(destination: JSResourceSelectView()) {
                 Text("JSResourceSelect")
             }
+            Button {
+                guard let url = try? hadesLogger.zipLogs() else {
```
Comment (sxiaojian88) [inline Maskbook/Scene/Debug/DebugView.swift:25]: Recommend put button action to a function

File: Maskbook/Utils/Logging.swift [hunk 2]
```diff
@@ -9,10 +9,16 @@
 import Foundation
 import MaskLog
 
+
```
Comment (sxiaojian88) [inline Maskbook/Utils/Logging.swift:12]: Recommend give a struct or class to this gloable var and function.

File: DevelopmentPod/MaskLog/MaskLog/Classes/HadesLogger/Handler/OSLogHandler.swift [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    subscript(metadataKey metadataKey: String) -> Logger.Metadata.Value? {
+        get {
+            return self.metadata[metadataKey]
+        }
+        set {
+            self.metadata[metadataKey] = newValue
+        }
+    }
+    
+    func log(
```
Comment (foxsin10) [inline DevelopmentPod/MaskLog/MaskLog/Classes/HadesLogger/Handler/OSLogHandler.swift:33]:
```suggestion
    func log(
    level: Logger.Level,
    message: Logger.Message,
    metadata: Logger.Metadata?,
    source: String,
    file: String = #file,
    function: String = #function,
    line: Int = #line
```

Comment (cythb) [reply DevelopmentPod/MaskLog/MaskLog/Classes/HadesLogger/Handler/OSLogHandler.swift:33]: You can't write it that way because the file here is passed in from where it was called, not the file where this method is located. The other two variables are the same case.

For example, if I call `log.debug("xxx")` on file A, at this point file should be file a. The default here doesn't make sense.

Comment (foxsin10) [reply DevelopmentPod/MaskLog/MaskLog/Classes/HadesLogger/Handler/OSLogHandler.swift:33]: 1. The file, function and line are passed down by the struct `Logger`.If there is some update for `Logger`, let's say replace `Logger` with some other object, keep the same signature will save some work.
2. Keep `#file`, `#function`, `#line` in default parameter, `OSLogHandler` can auto collect the file info.

Comment (cythb) [reply DevelopmentPod/MaskLog/MaskLog/Classes/HadesLogger/Handler/OSLogHandler.swift:33]: There is a reasons for not doing so.
If you write it this way, it is easy to make mistakes, here take `OSLogHandler` as an example.
`ProxyLogHandler` --> `OSLogHandler`
If this method has a default value, it is easy to cause the ProxyLogHandler not to use the parameters passed to it, but to pass its `#File` to the `OSLogHandler`.
For example, there is the following code.
```swift 
// AppDelegate.swift

// line 10
log.debug("test")
```
The log method of `ProxyLogHandler` has a file parameter of `AppDelegate.swift` instead of `ProxyLogHandler.swift`. If there was a default value, it would be easier to pass `ProxyLogHandler.swift` as a parameter to `OSLogHandler`.
In summary, I think it might be better to not have a default value here.

**Review by sxiaojian88 — APPROVED** [review]
